### ISSUE TITLE
Focus outline on Code Snippet is unstyled

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#2304]
When clicking in a code snippet, it appears that we have a un-styled focus indicator:

![image](https://user-images.githubusercontent.com/3901764/56085038-7f49cf80-5e02-11e9-9b43-f7ef2f1e01c5.png)

This is called (I believe) by `tabindex="0"` and `role="textbox"`, which I believe is moreso meant for an area where you can freely edit text.

**Comment by joshblack** [comment]
cc @dakahn do you agree on the textbox stuff?

**Comment by dakahn** [comment]
Yeah, exactly right. Wrong role. I can bundle these fixes in with some code snippet improvements that have been made recently to CCR. 👍🏽

**Comment by dakahn** [comment]
Looks like this was taken care of 😄

**Comment by janhassel** [comment]
I was just about to open an issue about the focus style on code snippet and found this issue.
In the current version (`carbon-components-react@7.9.2`) I can replicate the screenshot from @joshblack

![image](https://user-images.githubusercontent.com/28265588/73267165-bf830e80-41d8-11ea-81bd-2555ef95e5ca.png)

In the single-line codesnippet the focus style beyond the box, in the multi-line it's just unstyled:

![image](https://user-images.githubusercontent.com/28265588/73267224-e3deeb00-41d8-11ea-86e0-a3d1a6891f91.png)

Using Chrome v79 on macOs 10.14.6

**Comment by tw15egan** [comment]
Confirmed this is still present. @carbon-design-system/design , what should we do with this outline here?

**Comment by laurenmrice** [comment]
Can we do something like this:
![Artboard](https://user-images.githubusercontent.com/43969356/73307676-51444900-41e4-11ea-8ae8-b5755517a244.png)

**Comment by tw15egan** [comment]
With the way the component is set up, it doesn't seem like that is possible. 

<img width="819" alt="Screen Shot 2020-01-28 at 2 16 02 PM" src="https://user-images.githubusercontent.com/11928039/73310347-fbb66f00-41d8-11ea-8d30-b933bceca3a6.png">

<img width="796" alt="Screen Shot 2020-01-28 at 2 16 32 PM" src="https://user-images.githubusercontent.com/11928039/73310352-ff49f600-41d8-11ea-9603-4dfa63cbec59.png">


My recommendation would be to just remove the focus from the element since it is not interactive anyways. The focus will just go to the copy icon, then to show more.

**Comment by laurenmrice** [comment]
ok yeah then lets just do that, have the focus on the copy icon and then the show more only^ .

**Comment by tw15egan** [comment]
Focus styles have been removed from multi-line snippet, @emyarod is going to look into a possible cross-browser / os solution to the single line snippet in a separate PR

**Comment by emyarod** [comment]
just a proof of concept, still WIP but looking for feedback before I continue @laurenmrice @tw15egan 

here's a screenshot dump but we can go over this in more detail later on. every category below has 3 images each showing Windows 10, macOS with hidden scrollbars, and macOS with visible scrollbars. the first set is the as-is, the second set is a CSS only fix, and the third set includes a proof of concept that includes JS.

### as-is

##### default

![as-is1](https://user-images.githubusercontent.com/8265238/73406957-6345ea80-42bd-11ea-8e9e-aa3afd8d63d4.png)
![as-is1](https://user-images.githubusercontent.com/8265238/73406704-b0758c80-42bc-11ea-908c-50c8fa214157.png)
![as-is4](https://user-images.githubusercontent.com/8265238/73406728-bff4d580-42bc-11ea-8ecb-8dfc220831ff.png)

##### snippet focused

![as-is2](https://user-images.githubusercontent.com/8265238/73406990-7a84d800-42bd-11ea-813e-21e569ba70ce.png)
![as-is2](https://user-images.githubusercontent.com/8265238/73407018-88d2f400-42bd-11ea-8856-f2d76bd4fe98.png)
![as-is5](https://user-images.githubusercontent.com/8265238/73407019-88d2f400-42bd-11ea-85dc-cde40c0236d6.png)

##### copy button focused

![as-is3](https://user-images.githubusercontent.com/8265238/73407047-98ead380-42bd-11ea-88c1-c114e49c5769.png)
![as-is3](https://user-images.githubusercontent.com/8265238/73407042-96887980-42bd-11ea-871d-16e20e52361f.png)
![as-is6](https://user-images.githubusercontent.com/8265238/73407043-96887980-42bd-11ea-82f6-9fbfb47b8ff2.png)

### preliminary fix (no height adjustments)

##### default

![p1](https://user-images.githubusercontent.com/8265238/73407129-c6378180-42bd-11ea-9629-1c0daad33084.png)
![p1](https://user-images.githubusercontent.com/8265238/73407197-fda62e00-42bd-11ea-8c8e-390a880f97b5.png)
![p4](https://user-images.githubusercontent.com/8265238/73407198-fda62e00-42bd-11ea-8039-646795025ed0.png)

##### snippet focused

![p2](https://user-images.githubusercontent.com/8265238/73407260-2af2dc00-42be-11ea-900e-78c9d26c7445.png)
![p2](https://user-images.githubusercontent.com/8265238/73407269-33e3ad80-42be-11ea-84ab-be31424aa33e.png)
![p5](https://user-images.githubusercontent.com/8265238/73407270-33e3ad80-42be-11ea-9edd-887c20d35399.png)

##### copy button focused

![p3](https://user-images.githubusercontent.com/8265238/73407264-2f1ef980-42be-11ea-88e9-77bd8c2dd82d.png)
![p3](https://user-images.githubusercontent.com/8265238/73407273-37773480-42be-11ea-87c6-961020c56175.png)
![p6](https://user-images.githubusercontent.com/8265238/73407274-380fcb00-42be-11ea-9f37-cff8794bf27d.png)

### dynamically set height based on whether or not scrollbar is present

if the content is short and a scrollbar is not rendered, it will appear the same as the prelim fix screenshots above on macOS

##### default

![z1](https://user-images.githubusercontent.com/8265238/73407363-7c02d000-42be-11ea-9bb2-18ecb154406c.png)
![p1](https://user-images.githubusercontent.com/8265238/73407197-fda62e00-42bd-11ea-8c8e-390a880f97b5.png)
![z1](https://user-images.githubusercontent.com/8265238/73407400-9a68cb80-42be-11ea-8b8b-07514a18a7db.png)

##### snippet focused

![z2](https://user-images.githubusercontent.com/8265238/73407364-7c02d000-42be-11ea-8d11-13b1e1f3db54.png)
![p2](https://user-images.githubusercontent.com/8265238/73407269-33e3ad80-42be-11ea-84ab-be31424aa33e.png)
![z2](https://user-images.githubusercontent.com/8265238/73407481-dd2aa380-42be-11ea-924f-3fe92bf312ad.png)


##### copy button focused

![z3](https://user-images.githubusercontent.com/8265238/73407365-7c02d000-42be-11ea-9b13-f0383b160b98.png)
![p3](https://user-images.githubusercontent.com/8265238/73407273-37773480-42be-11ea-87c6-961020c56175.png)
![z3](https://user-images.githubusercontent.com/8265238/73407485-df8cfd80-42be-11ea-8e6b-2f6c2c467d8b.png)

**Comment by tw15egan** [comment]
@emyarod thanks for taking the time to put these screencaps together 🙏 Makes it very easy to illustrate the issue.

**Comment by emyarod** [comment]
if the "preliminary fix (no height adjustments)" solution is fine then I can continue with that, but the proof of concept with a bit of tweaking might be worth implementing

**Comment by tw15egan** [comment]
Personally, I think the preliminary fix (no height adjustments) looks great and could be implemented immediately it seems. I like that it stays the same height in both macOS versions.

**Comment by laurenmrice** [comment]
agreed, lets go for that ^
